src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java

```diff
@@ -220,6 +219,20 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                         commitMessages.getTagHotfixMessage(), gpgSignTag, properties);
             }
 
+            if (supportBranchName != null) {
```

Comment:
Why did you move merging after tagging? It will probably create tag on the wrong commit.

Reply:
I want to tag the hotfix branch, and not the production branch which never gets the changes merged in our use case.  Moving the merge to after tagging does that, and in the "normal' use of merging to production the tag is still created.  From my testing the change results in the tag being associated with the last commit of the branch instead of with the merge commit. 

Another approach would be to revert to merging before tagging, but not checking out production when 'skipMergeProdBranch' is specified. Tagging would then occur as normal.
Then immediately after the tagging check for the 'skipMergeProdBranch' (and supportBranchName is null) to ensure the expected branch is active for subsequent processing:
```suggestion
             if (skipMergeProdBranch && (supportBranchName == null)) {
                // git checkout master 
                gitCheckout(gitFlowConfig.getProductionBranch());
                }
```

Appreciate your insight

Reply:
Do you need to checkout production branch at all if `skipMergeProdBranch` is enabled?

Reply:
We need to switch off the hotfix branch so it can be deleted.  I think switching to production is safest as it aligns with the existing logic

Reply:
Yeah, it is seems that `if (skipMergeProdBranch && (supportBranchName == null))` is better than other alternatives where it is also needed to adjust currentVersion.

Go ahead!

Reply:
Can you also squash all the commits, after resolving this.

Reply:
will do, thanks

Reply:
Just pushed the latest.  Let me know if anything needs adjustment